toolchain/sem_ir/type.cpp

```diff
@@ -8,28 +8,45 @@
 
 namespace Carbon::SemIR {
 
-auto TypeStore::GetTypeIdForTypeConstantId(SemIR::ConstantId constant_id) const
-    -> SemIR::TypeId {
+// Verify that the constant value's type is `TypeType` (or an error).
+static void CheckConstantValueIsTypeType(File& file, ConstantId constant_id) {
```

Comment:
Should this be called `CheckConstantTypeIsTypeType` or `CheckConstantIsType` maybe, given we're checking the type of the constant, not the value?

Reply:
CheckConstantValueHasTypeOfTypeType? :/

IsType I think is wrong, since we called FacetTypes "constraint values" not "type values" though they are of type TypeType.

Reply:
> IsType I think is wrong, since we called FacetTypes "constraint values" not "type values" though they are of type TypeType.

A type is a value of type `type`. That includes facet types. Facet *values* are not types in general though (except in the special case where the facet type is exactly `type`).

Reply:
Maybe `CheckTypeOfConstantIsTypeType`? (Though I still think `CheckConstantIsType` or just `CheckIsType` would be fine.)

Reply:
Oh I do like that one best.

I just wanted to also draw attention to https://docs.google.com/presentation/d/1WJtP2yEBTKcKjvfYxnGJnN-sGmL4rCa9Xc7FWZcg0I8/edit?pli=1&slide=id.p#slide=id.p where we had FacetType in a higher typish-ness than type values, which we called constraint values. They are "types" in the implementation though, making this all the harder to speak about. I've been trying to only refer to things in the middle typish level as type values but maybe that is wrong and we should call both levels "type values".

Reply:
I maybe don't understand everything that diagram is trying to convey. Slides 2 and 3 of that presentation and #2360 reflect my understanding -- a type is any constant of type `type`, including facet type values (which are the types of facet values), class type values (which are the types of class values), function type values (which are the types of function values), and so on.